src/config.ts

```diff
@@ -0,0 +1,43 @@
+const hasDOM = typeof document !== 'undefined' && typeof window !== 'undefined';
+const scriptElement: HTMLOrSVGScriptElement | null = hasDOM ? document.currentScript : null;
+const globalConfig: Record<string, PrismConfig[keyof PrismConfig] | null> =
+	// @ts-ignore
+	globalThis.Prism?.constructor?.name === 'Object' ? globalThis.Prism : {};
+
+function getGlobalSetting (name: string) {
```

Comment:
Are these functions new? If not, where were they before?

Reply:
They are. They are from the `simplify` branch. I decided to introduce them here because they allow us to simplify `auto-start.ts` and prepare for adding support for other options.

Reply:
Here you go:

https://github.com/PrismJS/prism/blob/7c7c598d69ea5c669deea900597647ac22f8dfea/src/config.ts#L7-L23

and

https://github.com/PrismJS/prism/blob/7c7c598d69ea5c669deea900597647ac22f8dfea/src/config.ts#L25-L33